### PR TITLE
feat: add loadingCircle

### PR DESCRIPTION
**Pull request by groovywave**
[Issue No.07](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#7)
loadingを実装してみてください。 このようなローディング実装を作ってください。

[codesandbox](https://codesandbox.io/s/github/groovywave/JS_Practice/tree/feat/lesson07/lesson07?file=/main.js)

Latest Commit : eebf8e139c1fac56ef2cc426764c464b72994bb2

前回実装済
PromiseとsetTimeoutを使って３秒後にメールアイコン等が表示されるようにしました。

今回実装点
メールアイコン等が表示される前の３秒間ローディングアイコンが表示されるようにしました。

Concerns and areas to focus on
Can you read my code easily?

前回の課題から変更したファイル
・main.js

前回の課題から追加したファイル
・loading-circle.gif

### CODE REVIEW

File: lesson07/main.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-    {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
-    {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
-  ]; 
-  setTimeout(() => {
-    resolve(elements);
-  }, 3000);
-}).then((resolvedElements)=> {
-  const ul = document.getElementById("js-ul"); 
+const ul = document.getElementById("js-ul"); 
+const loadingCircle = document.createElement("img");
```
Comment (seeeen044) [inline lesson07/main.js:2]: IMO
It's more natural to be in a function.

File: lesson07/main.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+const loadingCircle = document.createElement("img"); 
+
+function renderCircle(){
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "loading-circle.gif";
+  ul.appendChild(loadingCircle);
+}  
+
+function renderContact(resolvedElements){
+  ul.removeChild(loadingCircle);
```
Comment (seeeen044) [inline lesson07/main.js:11]: IMO
It is better to separate it as a function to delete the loading.

**Review by seeeen044 — COMMENTED** [review]
I think it will be easy to read if you are conscious of letting one function do one job!

File: lesson07/main.js (hunk 2 above)
Comment (Saki-Ota) [reply lesson07/main.js:11]: I agree with sen-san, it is better to separate removing circle into an individual function instead on including within renderCircle()

File: lesson07/main.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-    resolve(elements);
-  }, 3000);
-}).then((resolvedElements)=> {
-  const ul = document.getElementById("js-ul"); 
+const ul = document.getElementById("js-ul"); 
+const loadingCircle = document.createElement("img"); 
+
+function renderCircle(){
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "loading-circle.gif";
```
Comment (kenmori) [inline lesson07/main.js:6]: MUST
I don't think it's a good idea to put such a value in the alt, you should look into what to set in the alt

File: lesson07/main.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+  ul.appendChild(loadingCircle);
+}  
+
+function renderContact(resolvedElements){
```
Comment (kenmori) [inline lesson07/main.js:10]: Q
What was your background in using the word contract?

File: lesson07/main.js (hunk 2 above)
Comment (groovywave) [reply lesson07/main.js:11]: Thank you sen-san and saki-san ;) 
I tried that like below and I wonder if it's o.k that the function has only a line.
function deleteCircle(){
  ul.removeChild(loadingCircle);
}
I'll read another code and learn it :)

File: lesson07/main.js (hunk 1 above)
Comment (groovywave) [reply lesson07/main.js:2]: Thank you sen-san :)
I tried that but 2 functions use ul and loadingCircle.
So I defined global constant value.

I'll read another code and learn it :)

File: lesson07/main.js (hunk 3 above)
Comment (groovywave) [reply lesson07/main.js:6]: Thank you teacher :)
I coded that it works when the loading-circle doesn't work.
I'll look into alt.

File: lesson07/main.js (hunk 4 above)
Comment (groovywave) [reply lesson07/main.js:10]: Thank you teacher :)
I meant contact.
I'll check that.

File: lesson07/main.js (hunk 2 above)
Comment (groovywave) [reply lesson07/main.js]: Thank you sen-san and saki-san :)
I fixed my code.
8a7e4213137c14761e81cd858b8b5c4575e5b859
I'm sorry about that I comiited evry fixing at once.

File: lesson07/main.js (hunk 4 above)
Comment (groovywave) [reply lesson07/main.js]: Thank you teacher :)
I renamed the function's name to renderIcons.

Comment (groovywave) [reply lesson07/main.js]: I fixed my code again.
0bb1711a1a92a7f74925ad6f8648ec43267cec8f

File: lesson07/main.js [hunk 5]
```diff
@@ -23,4 +29,20 @@ new Promise((resolve)=> {
       .insertAdjacentElement("afterbegin",img); 
   }; 
   ul.appendChild(fragment);
+}
+
+const getElements = new Promise((resolve)=> {
```
Comment (kenmori) [inline lesson07/main.js:34]: IMO
このファイルを読んでいるとElementはDOMに見えるのでgetMenusにしたいです

Comment (groovywave) [reply lesson07/main.js:34]: Thank you teacher :)

File: lesson07/main.js [hunk 6]
```diff
@@ -1,26 +1,48 @@
...
+  loadingCircle.alt = "ローディング画像";
+  loadingCircle.id = "loading-circle";
+  ul.appendChild(loadingCircle);
+}  
+
+function removeCircle(){
+  document.getElementById("loading-circle").remove();
+}
+
+function renderMenus(resolvedMenus){
```
Comment (kenmori) [inline lesson07/main.js:15]: renderMenusは渡って来た値が解決されたかどうかは知らないでもいいので
```suggestion
function renderMenus(values){

```

File: lesson07/main.js [hunk 7]
```diff
@@ -1,26 +1,48 @@
...
+  const menus = [ 
+    {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
+    {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
+  ]; 
+  setTimeout(() => {
+    resolve(menus);
+  }, 3000);
+})
+
+getMenus.then((resolvedMenus)=> {
```
Comment (kenmori) [inline lesson07/main.js:45]: ここもですね
```suggestion
getMenus.then((values)=> {

```

**Review by seeeen044 — APPROVED** [review]

File: lesson07/main.js (hunk 7 above)
Comment (groovywave) [reply lesson07/main.js:45]: Thank you teacher :）
I learned how to code.

**Pull request merged**